### PR TITLE
Wet/Dry Vac

### PR DESCRIPTION
**Pull request by Kurfursten**
This thing is really nifty. Cleans up spills of all kinds no problem - pulls lube or water right off the ground, and sucks up trash too. I'm not 100% satisfied with the sound effect if someone wants to send a new one, but it is mild enough that it doesn't get annoying hearing it over and over again.

You can use mousedrop onto the vac from the floor to do your sucking, or pull out the handle and click with it. Mousedrop onto a disposal bin/compactor to dump trash and onto a reagent container/sink to dump fluids. Handle automatically retracts if you try to leave it behind or go too far from the vac. It can hold 50 trash objects and 50u reagents (it gets 1u each time it cleans most decals/puddles and can get all kinds of fun stuff like lube, tomato juice, liquid butter, spiders, colorful reagent... it all depends on what you suck up!)

Sprites courtesy of @BurntDevil 

🆑 
* rscadd: Traders now sell a wet/dry vacuum for the rare janitor who DOESN'T want slippery floors.

### CODE REVIEW

File: code/game/objects/items/trader.dm [hunk 1]
```diff
@@ -752,3 +752,175 @@
...
+		if(!contents.len)
+			to_chat(usr,"<span class='warning'>\The [src] dry storage is already empty!</span>")
+			return
+		playsound(src, 'sound/effects/freeze.ogg', 25, 1) //this sounds like trash moving to me
+		for(var/obj/item/I in trash)
+			I.forceMove(O)
+			trash -= I
+		to_chat(usr, "<span class='notice'>You dump \the [src] dry contents into \the [O].</span>")
+
+/obj/structure/wetdryvac/MouseDropTo(atom/O, mob/user as mob)
```
Comment (ShiftyRail) [inline code/game/objects/items/trader.dm]: `as mob` isn't needed.

File: code/game/objects/items/trader.dm [hunk 2]
```diff
@@ -752,3 +752,175 @@
...
+	if(!can_use(user,O))
+		return
+	whrr(get_turf(O))
+
+/obj/structure/wetdryvac/proc/whrr(var/turf/T)
```
Comment (ShiftyRail) [inline code/game/objects/items/trader.dm:875]: >the proc name

:pensive:

File: code/game/objects/items/trader.dm [hunk 3]
```diff
@@ -752,3 +752,175 @@
...
+	if(!T)
+		return
+	playsound(src, 'sound/effects/vacuum.ogg', 25, 1)
+	for(var/obj/effect/decal/cleanable/C in T)
+		var/datum/reagent/R = C.reagent
```
Comment (ShiftyRail) [inline code/game/objects/items/trader.dm]: The reagent in the cleanable seem to be strings, not types. You would need to create a reagent datum from the string of the decal, I think.
https://github.com/vgstation-coders/vgstation13/blob/f9a648cab932440204222263871df888c6e8e6b5/__DEFINES/reagents.dm#L186

Comment (Kurfursten) [reply code/game/objects/items/trader.dm]: add_reagent wants a string, the real mistake I made here was having R be a reagent datum at all, it should just be a string

**Comment by tuypo1** [comment]
>just as the crew thinks a spider situation is under control a rouge janitor starts it up again

Fun

**Review by ShiftyRail — APPROVED** [review]

**Pull request merged**